src/rust/engine/src/externs/interface.rs

```diff
@@ -685,12 +686,20 @@ async fn workunit_to_py_value(
       ),
     ];
 
-    if let Some(parent_id) = workunit.parent_id {
-      dict_entries.push((
-        externs::store_utf8(py, "parent_id"),
-        externs::store_utf8(py, &format!("{}", parent_id)),
-      ));
+    let parent_ids = workunit
+      .parent_ids
+      .into_iter()
+      .map(|parent_id| externs::store_utf8(py, &parent_id.to_string()))
+      .collect::<Vec<_>>();
+
+    if has_parent_ids {
+      // TODO: Remove the single-valued `parent_id` field around version 2.16.0.dev0.
```

Comment:
cc @asherf

Reply:
As an aside, the streaming workunit handler has a "version" attribute that could be incremented to make it easier to account for changes such as this.